scripts/services/migrations.ts

```diff
@@ -1,23 +1,154 @@
+                bitcoind: matches.shape({ type: matches.any }, ["type"]),
+              }).test(config)
+            ) {
+              config.bitcoind.type = "internal";
+            }
+            return config;
+          },
+          false,
+          { version: "0.13.3.2", type: "down" },
+        ),
+      },
+      "0.14.2": {
+        up: compat.migrations.updateConfig(
+          (config) => {
+            if (
+              matches.shape({
+                bitcoind: matches.shape({ type: matches.any }, ["type"]),
+              }).test(config)
+            ) {
+              if ( config.bitcoind.type == "external"){ config.bitcoind.type = "internal-proxy" }
+            }
+            if (
+              matches.shape({
+                tor: matches.shape({
+                  "use-tor-only": matches.any,
+                  "stream-isolation": matches.any,
+                }),
+              }).test(config)
+            ) {
+              delete config.tor["use-tor-only"];
+              delete config.tor["stream-isolation"];
+            }
+            return config;
+          },
+          false,
+          { version: "0.14.2", type: "up" },
+        ),
+        down: compat.migrations.updateConfig(
+          (config) => {
+            if (
+              matches.shape({
+                tor: matches.shape({
+                  "use-tor-only": matches.any,
+                  "stream-isolation": matches.any,
+                }),
+              }).test(config)
+            ) {
+              delete config.tor["use-tor-only"];
+              delete config.tor["stream-isolation"];
+            }
+            return config;
+          },
+          false,
+          { version: "0.14.2", type: "down" },
+          ),
+      },
+      "0.14.2.1": {
+        up: compat.migrations.updateConfig(
+          (config) => {
+            if (
+              matches.shape({
+                bitcoind: matches.shape({ type: matches.any }, ["type"]),
+              }).test(config)
+            ) {
+              if ( config.bitcoind.type == "none"){ config.bitcoind.type = "internal-proxy" }
+            }
+            if (
+              matches.shape({
+                "watchtower-enabled": matches.any,
+                "watchtower-client-enabled": matches.any,
+              }).test(config)
+            ) {
+              delete config["watchtower-enabled"];
+              delete config["watchtower-client-enabled"];
+            }
+            return config;
+          },
+          false,
+          { version: "0.14.2.1", type: "up" },
+        ),
+        down: compat.migrations.updateConfig(
+          (config) => {
+            return config;
+          },
+          false,
+          { version: "0.14.2.1", type: "down" },
+        ),
+      },
+      "0.15.0": {
+        up: compat.migrations.updateConfig(
+          (config) => {
+            if (
+              matches.shape({
+                "watchtowers": matches.any,
```

Comment:
Not a big fan of any, but it does seem to reduce code complexity. I don't know the answer